resilience4j-all/src/main/java/io/github/resilience4j/decorators/Decorators.java

```diff
@@ -67,18 +58,6 @@ static DecorateRunnable ofRunnable(Runnable runnable) {
         return new DecorateCallable<>(callable);
     }
 
-    static <T> DecorateCheckedSupplier<T> ofCheckedSupplier(CheckedFunction0<T> supplier) {
```

Comment:
I am not sure but I think we assumed that 2.0 is going to have few contract breaking changes since we have made similar changes in previous PRs like `VavrCircuitBreaker`, `VavrRatelimiter` etc. @RobWin correct me if I am wrong.

Reply:
Yes, breacking changes for 2.0 are fine

Reply:
But I like the idea to add our own CheckedRunnable and CheckedFunction interfaces is a good idea. That means, if someone just uses Lambas, he does not even have to migrate the code.

Reply:
But we can do this in another PR

Reply:
This way we could keep existing tests.

Reply:
Since it's Phase-3, we are already too late ;) 
We can copy the tests later? wdyt?

Reply:
I would add interfaces now and keep tests at the same time.

Reply:
Meanwhile I was working on the next PR and have added  our own `CheckedFunction` `CheckedRunnable` etc. It would be easy for me to do it in next PR.

Reply:
@hexmind @RobWin let me know if that's fine. I should have foreseen the need but while working on next the PR I realised the need of having our own checkedFunction and other checked interfaces. hence I have added the same in upcoming PR.